src/test/java/org/gridsuite/study/server/NetworkModificationTreeTest.java

```diff
@@ -602,23 +608,43 @@ public void testGetLastParentNodeBuilt() throws Exception {
         assertEquals(node7.getId(), networkModificationTreeService.doGetLastParentNodeBuilt(node3.getId()));
     }
 
-    private void createNode(UUID studyUuid, AbstractNode parentNode, AbstractNode newNode) throws Exception {
+    private void createNode(UUID studyUuid, AbstractNode parentNode, NetworkModificationNode newNode) throws Exception {
```

Comment:
Maybe we should keep the generic AbstractNode type here no ?

Reply:
No 
This private method only called for a `NetworkModificationNode`